### ISSUE TITLE
The font size returned by tesseract is not accurate

### ISSUE THREAD

**Issue by Sharcoux** [naptha/tesseract.js#707]
**Describe the bug**

The font_size property returned by the ocr is not accurate.

**To Reproduce**

recognize the following image: ![Capture d’écran du 2023-01-06 18-59-36](https://user-images.githubusercontent.com/797545/211071521-fdbffb85-3aab-418c-84da-59f6357e7baf.png)

If you log the result of `data.lines.map(word => console.log(word.text, word.font_size))`, you will receive somthing like that:

![image](https://user-images.githubusercontent.com/797545/211072144-7840c243-6326-42b2-88da-0088a133bd01.png)

As you can see, sometimes the result is 3, sometimes it's 5. But all the characters are using the same font size

**Expected behavior**

An accurate answer would be "4"

**Screenshots**
Here is the consequence:

![image](https://user-images.githubusercontent.com/797545/211072543-402249a0-f59f-435c-b72d-015b85348a12.png)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 22.04
 - Browser [e.g. chrome, safari] Firefox
 - Version [e.g. 22] 108.0.1

**Smartphone (please complete the following information):**

**Additional context**

**Comment by Balearica** [comment]
The `font_size` attribute returned by Tesseract is indeed often incorrect.  Closing as this issue as this is a bug with Tesseract and not specific to the Javascript/Webassembly port or this library, so troubleshooting and fixing is not within the scope of this repository.  

If you explain what you're working on I may still be able to help, as it is still possible to determine font size manually by using the size of character bounding boxes.  This is how font size is implemented in [scribeocr.com](https://scribeocr.com/), which is able to precisely replicate the text from images using output from Tesseract.js.

**Comment by Sharcoux** [comment]
@Balearica I'm not entirely sure about how you compute the size.

Ok, here is the thing:

- I can estimate the word height with: `const height = word.bbox.y1 - word.bbox.y0`
- I use `@pdf-lib/fontkit` to be able to estimate the correct font size to match the drawing: `const fontSize = font.sizeAtHeight(height)`
- I can now compute the font size with: `word.font_size * 3.3 * height / fontHeight`. This is the most precise formula I found. Basically, `height / fontHeight` gives me the coefficient to convert a height into a font size, and Tesseract font_size value is around 3.3 times smaller than the actual font height. 

I don't see exactly how I can calculate this without the word font-size. Because, for instance, a word like `moon` will have a box size smaller than `fog` because of the f and g increasing the box. But if you succeeded, I'm interested to know how.

**Comment by Balearica** [comment]
Yes, different letters have different heights, so the calculation needs to make use of character-level bounding boxes rather than word-level bounding boxes.  For [scribeocr.com](https://scribeocr.com/) we define multiple classes of character:  ascenders, descenders, and x-height (e.g. `x`, `a`, `c`, etc.).  The median size of each class of characters is then calculated for each line, and those are used to determine the font size.  If you wanted to do a simplified implementation, I would take the median height of the x-height characters on each line ("a", "c", "e", "m", "n", "o", "r", "s", "u", "v", "w", "x", "z") and use that to determine font size.

**Comment by Sharcoux** [comment]
I could get some accurate results using this, but the issue is that the bbox from tesseract is also sometimes completely broken, while the font size is ok. So, in the end, my implementation relying on the font_size was generally more accurate than the new implementation. I don't understand how you can accurately rely on the bbox only and still return accurate results... :/

**Comment by Sharcoux** [comment]
@Balearica There is something I don't understand. Why do I get so low values for font_size? Check out [this guy](https://github.com/tesseract-ocr/tesseract/issues/3988#issuecomment-1376775267): with the exact same image, he gets font sizes from 18 to 23 while I get font size from 3 to 5...

**Comment by Balearica** [comment]
The legacy model (oem mode `0`) is going to produce accurate bounding boxes, whereas the LSTM model does not.  This is simply due to how those models work (the legacy model works by identifying and classifying individual characters).  The inaccurate LSTM model bounding boxes are a known issue--if you look in the Tesseract GitHub there are several open issues about this and at least one open PR with attempted fixes. 

I have found that the legacy model produces bounding boxes that are generally very accurate--attached is an image showing an OCR text overlay from [scribeocr.com](https://scribeocr.com/) (text size and positioning is based on the character-level metrics created by the Tesseract Legacy model). 

![optimization_comp1b1](https://user-images.githubusercontent.com/100809261/211632013-0d9f817d-db5a-45d1-837a-b2f714109412.png)

Regarding why `font_size` makes no sense, the calculation Tesseract uses for `font_size` is simply incorrect and cannot be relied upon.  Unfortunately I forgot exactly what was wrong with it, however at some point I reviewed the formula in the Tesseract source code and found that it was not correct.
